hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -850,7 +851,7 @@ genCollectComTx = do
   commits <- genCommits ctx txInit
   cctx <- pickChainContext ctx
   let (committedUTxO, stInitialized) = unsafeObserveInitAndCommits cctx txInit commits
-  pure (cctx, committedUTxO, stInitialized, collect cctx stInitialized)
+  pure (cctx, committedUTxO, stInitialized, addChangeOutput $ collect cctx stInitialized)
```

Comment:
I don't like this. Why would we need another output (I assume) to make the generated transaction valid?

The validator should be ignoring any other outputs than the first one when we simplify access based on the assumption of this PR.

Reply:
This is no longer needed.